### PR TITLE
Add unit tests for src/main/java/app/fitbuddy/operation/service/NewUserService.java

### LINKED ISSUE #110: Add unit tests for `src/main/java/app/fitbuddy/operation/service/NewUserService.java`

### PR DESCRIPTION
**Pull request by dmitriydb**
## Summary 

1. The class `app.fitbuddy.jpa.entity.DefaultExercise` does not have `@RequiredArgsConstructor` Lombok annotation or the constructor with `id` and `name` parameters so it's kinda hard to create dummy instances for testing. 
I was thinking about extending it with another test class that has such constructor (decorator pattern) for testing purpose but maybe it's a bad idea
Is it possible to add all-args constructor to `DefaultExercise` and if it is could I open the corresponding issue?

2. What should the `NewUserService` bean do when `addDefaultExercises` method is invoked with appUserId null or negative value? I haven't write the corresponding tests yet

## Close issue(s) 
Close #110

### CODE REVIEW

**Comment by mepox** [comment]
Hi @dmitriydb ! Thank you for your contribution again 👍 I appreciate you took your time to work on this.

> 1. The class `app.fitbuddy.jpa.entity.DefaultExercise` does not have `@RequiredArgsConstructor` Lombok annotation or the constructor with `id` and `name` parameters so it's kinda hard to create dummy instances for testing. 
I was thinking about extending it with another test class that has such constructor (decorator pattern) for testing purpose but maybe it's a bad idea
Is it possible to add all-args constructor to `DefaultExercise` and if it is could I open the corresponding issue?

I can't remember what was my reasoning to not use the `@RequiredArgsConstructor` on entities. However there are TestHelper classes in `/src/test/java/app/fitbuddy/testhelper/` to create mocks for entities.

> 2. What should the `NewUserService` bean do when `addDefaultExercises` method is invoked with appUserId null or negative value? I haven't write the corresponding tests yet

Good question, as I forget to add checks for the appUserId... 😅 I think the best would be to throw a new FitBuddyException for now with a message "Internal server error - appUserId is not correct"? Can you add these checks to the `NewUserService` please?

**Comment by dmitriydb** [comment]
> However there are TestHelper classes in /src/test/java/app/fitbuddy/testhelper/ to create mocks for entities.

OK, refactored the test code to use your implementation instead of creation my own.
Also I've added isEqual methods into DefaultExerciseHelper.

> Can you add these checks to the NewUserService please
Done

File: src/test/java/app/fitbuddy/testhelper/DefaultExerciseTestHelper.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package app.fitbuddy.testhelper;
 
+import app.fitbuddy.dto.ExerciseDto;
 import app.fitbuddy.jpa.entity.DefaultExercise;
+import app.fitbuddy.jpa.entity.Exercise;
```
Comment (mepox) [inline src/test/java/app/fitbuddy/testhelper/DefaultExerciseTestHelper.java:5]: Unused import in Line 5

File: src/test/java/app/fitbuddy/operation/service/NewUserServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
+package app.fitbuddy.operation.service;
+
+import app.fitbuddy.dto.ExerciseDto;
+import app.fitbuddy.exception.FitBuddyException;
+import app.fitbuddy.jpa.entity.DefaultExercise;
+import app.fitbuddy.jpa.repository.DefaultExerciseCrudRepository;
+import app.fitbuddy.jpa.service.crud.ExerciseCrudService;
+import app.fitbuddy.testhelper.DefaultExerciseTestHelper;
+import org.junit.jupiter.api.BeforeEach;
```
Comment (mepox) [inline src/test/java/app/fitbuddy/operation/service/NewUserServiceTest.java:9]: Unused import in Line 9

File: src/test/java/app/fitbuddy/operation/service/NewUserServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+			newUserService.addDefaultExercises(null);
+		});
+	}
+
+	@Test
+	public void newUser_whenAppUserIsNegative_shouldThrowException() {
+		assertThrows(FitBuddyException.class, () -> {
+			newUserService.addDefaultExercises(-7);
+		});
+	}
```
Comment (mepox) [inline src/test/java/app/fitbuddy/operation/service/NewUserServiceTest.java:76]: We could shorten this part by using `@ParameterizedTest`. Then use `@NullSource` annotation to provide a null value and use `@ValueSource` to provide the exact values.

```
@ParameterizedTest
@NullSource
@ValueSource(ints = { -7 })
public void example(Integer input) { ...
```

**Review by mepox — CHANGES_REQUESTED** [review]
@dmitriydb Thanks for your previous refactor. 👍 

The PR looks good. I only found a few small issues, please see my comments.

Once you apply those changes, I will merge this PR 🎉  Thanks

File: src/test/java/app/fitbuddy/operation/service/NewUserServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Value;
```
Comment (mepox) [inline src/test/java/app/fitbuddy/operation/service/NewUserServiceTest.java:18]: Unused import in Line 18

**Review by mepox — CHANGES_REQUESTED** [review]

**Comment by mepox** [comment]
@dmitriydb 
Unfortunately, I found an other unused import, see my comment. I guess you just added it now by mistake.

**Comment by dmitriydb** [comment]
@mepox 
Sorry, I fixed that. Is it better now?
I hope you're not annoyed yet 😄  I have little experience with collaboration

I see you're separating import sections with empty line, I've tried to replicate that in `NewUserServiceTest` too.

**Review by mepox — APPROVED** [review]

**Comment by mepox** [comment]
> @mepox Sorry, I fixed that. Is it better now? I hope you're not annoyed yet 😄 I have little experience with collaboration
> 
> I see you're separating import sections with empty line, I've tried to replicate that in `NewUserServiceTest` too.

@dmitriydb 
Thank you for your contribution and for applying the changes. 🥇  The PR looks good now, it will be merged soon. 🎉 

No worries, I am not annoyed 👍 

To be honest, my IDE automatically add and separate the imports 😅 But thanks for trying to replicate it and adding them nicely. 👍

**Pull request merged**